### PR TITLE
feat: Creates skeleton for `mongodbatlas_employee_access_grant` resource and singular data source

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Creates skeleton for `mongodbatlas_employee_access_grant` resource and singular data source.

This will create a "no-op" resource, i.e., a resource that is valid, check attributes, can be created / updated / deleted, etc., but doesn't call Atlas.

Example of Terraform file that this PR allows:
```
resource "mongodbatlas_employee_access_grant" "access" {
  project_id   = "123"
  cluster_name = "cluster"

  grant_type      = "type"
  expiration_time = "time"
}

data "mongodbatlas_employee_access_grant" "ds_access" {
  project_id   = "123"
  cluster_name = "cluster"
}
```

Link to any related issue(s): CLOUDP-269499

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/zclconf/go-cty v1.15.0
 	go.mongodb.org/atlas v0.37.0
 	go.mongodb.org/atlas-sdk/v20240530005 v20240530005.0.0
-	go.mongodb.org/atlas-sdk/v20240805003 v20240805003.1.0
+	go.mongodb.org/atlas-sdk/v20240805003 v20240805003.1.1-0.20240905175558-a2dcadd2b3dd
```
Comment (lantoli) [inline go.mod:27]: SDK Preview, won't be merged to master

File: internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go [hunk 2]
```diff
@@ -260,7 +260,7 @@ func resourceCloudProviderAccessAuthorizationStateUpgradeV0(ctx context.Context,
 }
 
 func authorizeRole(ctx context.Context, client *admin.APIClient, d *schema.ResourceData, projectID string, targetRole *admin.CloudProviderAccessRole) diag.Diagnostics {
-	req := &admin.CloudProviderAccessRole{
+	req := &admin.CloudProviderAccessRoleRequestUpdate{
```
Comment (lantoli) [inline internal/service/cloudprovideraccess/resource_cloud_provider_access_authorization.go:263]: minimum changes to get SDK Preview compile, won't be merged to master

File: internal/service/employeeaccessgrant/resource.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	resp.Schema = ResourceSchema(ctx)
+}
+
+func (r *employeeAccessGrantRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var tfPlan TFEmployeeAccessGrantModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, tfPlan)...)
```
Comment (lantoli) [inline internal/service/employeeaccessgrant/resource.go:37]: minimum operation code to have a "no-op" resource

**Review by marcosuma — APPROVED** [review]

File: internal/service/employeeaccessgrant/resource.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+func (r *employeeAccessGrantRS) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var tfPlan TFEmployeeAccessGrantModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfPlan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, tfPlan)...)
+}
+
+func (r *employeeAccessGrantRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
```
Comment (EspenAlbert) [inline internal/service/employeeaccessgrant/resource.go:52]: Will it be possible to run `terraform destroy`?

**Review by EspenAlbert — APPROVED** [review]
LGTM

File: internal/service/employeeaccessgrant/resource.go (hunk 4 above)
Comment (lantoli) [reply internal/service/employeeaccessgrant/resource.go:52]: I removed the resource from the file, applied, and went away.   

I suppose it'll be similar running terraform destroy ?

**Pull request merged**